pkg/cmd/server/kubernetes/node_config_test.go

```diff
@@ -157,3 +160,49 @@ func TestProxyConfig(t *testing.T) {
 	}
 
 }
+
+func TestBuildCloudProviderFake(t *testing.T) {
+	providerName := "fake"
+	cloudprovider.RegisterCloudProvider(providerName, func(config io.Reader) (cloudprovider.Interface, error) {
+		return &fake.FakeCloud{}, nil
+	})
+
+	server := kubeletoptions.NewKubeletServer()
+	server.CloudProvider = providerName
+
+	cloud, err := buildCloudProvider(server)
```

Comment:
@smarterclayton are you looking for a test that ensures that BuildKubernetesNodeConfig ends up with the cloud provider filled in correctly?